generators/entity/index.js

```diff
@@ -1000,6 +1040,50 @@ module.exports = EntityGenerator.extend({
 
     prompting: {
         /* pre entity hook needs to be written here */
+        askForMicroserviceJson: function(){
+            if(this.applicationType != 'gateway' || this.useConfigurationFile) {
+                return;
+            }
+
+            var cb = this.async();
+
+            var prompts = [
+                {
+                    type: 'confirm',
+                    name: 'useMicroserviceJson',
+                    message: 'Do you want to generate this entity from an existing microservice?',
+                    default: true
+                },
+                {
+                    when: function(response) {
+                        return response.useMicroserviceJson == true;
+                    },
+                    type: 'input',
+                    name: 'microservicePath',
+                    message: 'Enter the absolute path to the microservice root directory:',
```

Comment:
you can get path of [current directory ](http://yeoman.io/authoring/file-system.html)from where the generator is called this way `this.destinationRoot()` which is what you have assigned to `this.rootDir` so if I invoke `yo jhipster:entity Foo` from `/usr/someDir/myGatewayDir` `this.rootDir` should be the same and we can use a relative path to this folder as well like `../myMicroserviceOne` which is in `/usr/someDir/myMicroserviceOne`